streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -719,8 +718,7 @@ private KafkaStreams(final InternalTopologyBuilder internalTopologyBuilder,
         }
 
         // create the stream thread, global update thread, and cleanup thread
-        threads = new StreamThread[numStreamThreads];
-
+        threads = new LinkedList<>();
```

Comment:
Should this collection be threadsafe? (or are all accesses inside synchronized blocks anyway?)

Reply:
They are not all in synchronized blocks so I think it probably should be thread safe

Reply:
Currently only the constructor adds elements to the list. All other accesses afterwards are read-only. So, we would not need a synchronized list as far as I see. When we implement the add and remove streams thread APIs, we probably need synchronization. My proposal is to leave it a synchronized list now just in case we forget to think about it afterwards and then to reconsider how we synchronize the accesses.